Assets/MRTK/SDK/Features/UX/Scripts/InteractiveElement/Events/EventReceiverManager.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License
+
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+namespace Microsoft.MixedReality.Toolkit.UI.Interaction
+{
+    /// <summary>
+    /// Manages the events contained in states within an Interaction Element.
+    /// </summary>
+    public class EventReceiverManager
+    {
+        /// <summary>
+        /// Constructor for the event receiver manager. 
+        /// </summary>
+        /// <param name="interactiveStateManager">The state manager for this event receiver manager</param>
+        internal EventReceiverManager(StateManager interactiveStateManager)
+        {
+            stateManager = interactiveStateManager;
+        }
+
+        // The state manager for this event receiver manager
+        private StateManager stateManager = null;
+
+        /// <summary>
+        /// List of active event receivers for the state events.
+        /// </summary>
+        public List<BaseEventReceiver> EventReceivers { get; protected set; } = new List<BaseEventReceiver>();
+
+        /// <summary>
+        /// Initialize the event receivers for each state that has an existing event configuration.
+        /// </summary>
+        internal void InitializeEventReceivers()
+        {
+            foreach (InteractionState state in stateManager.States)
+            {
+                // Initialize Event Configurations for each state if the configuration exists
+                if (IsEventConfigurationValid(state.Name))
+                {
+                    // If an interactive element component is created via script instead of initialized in the inspector,
+                    // an instance of the event config scriptable needs to be created
+                    if (state.EventConfiguration == null)
+                    {
+                        state.EventConfiguration = CreateEventConfigurationInstance(state.Name);
+                    }
+
+                    // Initialize runtime event receiver classes for states that have an event configuration 
+                    InitializeEventReceiver(state.Name);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Invoke a state event with optional event data.
+        /// </summary>
+        /// <param name="stateName">The name of the state</param>
+        /// <param name="eventData">The event data for the state event</param>
+        public void InvokeStateEvent(string stateName, BaseEventData eventData = null)
+        {
+            BaseEventReceiver receiver = EventReceivers.Find((eventReceiver) => eventReceiver.Name.StartsWith(stateName));
```

Comment:
Probably won't be an issue since the list won't be that long, but you could consider using a Dictionary<string, BaseEventReceiver> instead of list for faster lookup, especially if this or other code that does list lookup like this runs every frame.

Reply:
That is more efficient, great suggestion.  Changed the list to a dictionary.